File: JJB/1128/하노이의_탑.cpp
#include <string>
#include <vector>

using namespace std;

void dfs(int n, int from, int to, int aux, vector<vector<int>> &result)
{
    if(n==1)
    {
        result.push_back({from, to});
        return;
    }
    
    dfs(n-1, from, aux, to, result);
    result.push_back({from, to});
    dfs(n-1, aux, to, from, result);
}

vector<vector<int>> solution(int n) 
{
    vector<vector<int>> answer;
    
    dfs(n, 1, 3, 2, answer);
    
    return answer;
}

File: JJB/1128/하노이의_탑.py
def hanoi_tower(n, from_rod, to_rod, aux_rod, result):
    if n == 1:
        result.append([from_rod, to_rod])
        return
    hanoi_tower(n - 1, from_rod, aux_rod, to_rod, result)
    result.append([from_rod, to_rod])
    hanoi_tower(n - 1, aux_rod, to_rod, from_rod, result)

def solution(n):
    answer = []
    hanoi_tower(n, 1, 3, 2, answer)
    return answer